erddapy/erddapy.py

```
@@ -147,12 +192,22 @@ def __init__(
         self.requests_kwargs: Dict = {}
         self.auth: Optional[tuple] = None
         self.variables: Optional[ListLike] = None
+        self.dim_names: Optional[ListLike] = None
 
         # Caching the last `dataset_id` and `variables` list request for quicker multiple accesses,
         # will be overridden when requesting a new `dataset_id`.
         self._dataset_id: OptionalStr = None
         self._variables: Dict = {}
 
+    def griddap_initialise(self):
+        if self.protocol == "griddap":
```

Should probably throw an error is the protocol is not griddap. What do you think?